src/checkbox/checkbox.ts

```diff
@@ -0,0 +1,102 @@
+import {Component, ElementRef, EventEmitter, forwardRef, Input, Output} from '@angular/core';
+import {ControlValueAccessor, NG_VALUE_ACCESSOR} from '@angular/forms';
+
+const NGB_CHECKBOX_VALUE_ACCESSOR = {
+  provide: NG_VALUE_ACCESSOR,
+  useExisting: forwardRef(() => NgbCheckbox),
+  multi: true
+};
+
+let checkbox_id_generator = 1;
+
+/**
+ * A component that allows the user to edit boolean values.
+ */
+@Component({
+  selector: 'ngb-checkbox',
+  host: {'[class.custom-control]': 'true', '[class.custom-checkbox]': 'true'},
+  template: `
+    <input
+      type="checkbox"
+      class="custom-control-input"
+      [disabled]="disabled || readonly"
+      [checked]="value"
+      (change)="_valueChange($event)"
+      (blur)="_blur()"
+      [attr.id]="id"
```

Comment:
`id` is mapped, so we could use just:
```html
[id]="id"
```
https://angular.io/guide/template-syntax#html-attribute-vs-dom-property